pymess/backend/dialer/__init__.py

```diff
@@ -0,0 +1,203 @@
+        :param template: template object from which content of the message was create
+        :param dialer_kwargs: extra attributes that will be stored to the message
+        """
+        message = self.create_message(recipient, content, related_objects, tag, template, **dialer_kwargs)
+        self.publish_message(message)
+        return message
+
+    def bulk_send(self, recipients, content, related_objects=None, tag=None, template=None, **dialer_kwargs):
+        """
+        Send more Dialer messages in one bulk
+        :param recipients: list of phone numbers of recipients
+        :param content: text content of the messages
+        :param related_objects: list of related objects that will be linked with the SMS message using generic
+        relation
+        :param tag: string mark that will be saved with the message
+        :param template: template object from which content of the message was create
+        :param dialer_kwargs: extra attributes that will be stored with messages
+        """
+        messages = [
+            self.create_message(recipient, content, related_objects, tag, template, **dialer_kwargs)
+            for recipient in recipients
+        ]
+        self.publish_messages(messages)
+        return messages
+
+    def get_initial_dialer_state(self, recipient):
+        """
+        returns initial state for logged Dialer instance.
+        :param recipient: phone number of the recipient
+        """
+        return DialerMessage.STATE.WAITING
+
+    def _update_dialer_states(self, messages):
+        """
+        If Dialer sender provides check Dialer delivery this method can be overridden.
+        :param messages: messages which state will be updated
+        """
+        raise NotImplementedError('Check Dialer state is not supported with the backend')
+
+    def bulk_check_dialer_states(self):
+        """
+        Method that find messages that is not in the final state and updates its states.
+        """
+        messages_to_check = DialerMessage.objects.filter(state=DialerMessage.STATE.SENDING)
+        if messages_to_check.exists():
+            self._update_dialer_states(messages_to_check)
+
+        idle_dialer_messages = DialerMessage.objects.filter(
+            state=DialerMessage.STATE.SENDING,
+            created_at__lt=timezone.now() - timedelta(minutes=settings.DIALER_IDLE_MESSAGES_TIMEOUT_MINUTES)
+        )
+        if settings.DIALER_LOG_IDLE_MESSAGES and idle_dialer_messages.exists():
+            LOGGER.warning('{count_dialer} Dialer message is more than {timeout} minutes in state "SENDING"'.format(
+                count_dialer=idle_dialer_messages.count(), timeout=settings.DIALER_IDLE_MESSAGES_TIMEOUT_MINUTES
+            ))
+
+        if settings.DIALER_SET_ERROR_TO_IDLE_MESSAGES:
+            idle_dialer_messages.update(
+                state=DialerMessage.STATE.ERROR, error=ugettext('timeouted')
+            )
+
```

Comment:
ještě tu budeš muset řešit update stavu zprávy, jako tomu je na SMS. Na to se bude muset přidat command (check sms delivery)